packages/vitest-pool-workers/src/config/index.ts

```diff
@@ -161,15 +176,43 @@ function ensureWorkersConfig<T extends UserConfig>(config: T): T {
 	return config;
 }
 
+export function defineWorkersConfig(
+	config: WorkersUserConfigExport
+): WorkersUserConfigExport
+export function defineWorkersConfig(
+	config: Promise<WorkersUserConfigExport>
+): Promise<WorkersUserConfigExport>
+export function defineWorkersConfig(
+	config: ConfigFn<WorkersUserConfigExport>
+): ConfigFn<WorkersUserConfigExport>
 export function defineWorkersConfig(
 	config: AnyConfigExport<WorkersUserConfigExport>
 ): AnyConfigExport<WorkersUserConfigExport> {
+	if (typeof config === 'function') {
```

Comment:
Type guard is needed because `mayAnyConfigExport` is also overloaded.
I could remove `mayAnyConfigExport` and write its body directly into `defineWorkersConfig` and `defineWorkersProject` to avoid many overload signatures, but  I didn't know the original intent of `mapAnyConfigExport` so I decided to keep it